apps/website/src/server/notifications.ts

```diff
@@ -0,0 +1,132 @@
+import { notificationCategories } from "@/config/notifications";
+
+import { prisma } from "@/server/db/client";
+import { callEndpoint } from "@/server/utils/queue";
+
+import type { CreatePushesOptions } from "@/pages/api/notifications/batched-create-notification-pushes";
+import type { RetryPushesOptions } from "@/pages/api/notifications/batched-retry-notification-pushes";
+
+const PUSH_BATCH_SIZE = 2; // TODO: make configurable? env file?
```

Comment:
TODO: definitely make this configurable so we can increase this in production. I dont really have an idea how to figure out a good value here though. Probably will be trial and error at first, but it would be nice to measure the time each of the routes/handlers/tasks takes and scale them appropriately and according to the limits by vercel.